uint32 fault_code
uint32 temp
bool limits_set